src/Sentry/Internal/ProcessInfo.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Sentry.Extensibility;
+
+namespace Sentry.Internal
+{
+    internal class ProcessInfo
+    {
+        internal static ProcessInfo? Instance;
+
+        /// <summary>
+        /// When the code was initialized.
+        /// </summary>
+        internal DateTimeOffset StartupTime { get; set; }
+
+        /// <summary>
+        /// When the device was initialized.
+        /// </summary>
+        internal DateTimeOffset BootTime { get; }
+
+        private SentryOptions _options { get; }
+
+        internal ProcessInfo(SentryOptions options)
+        {
+            _options = options;
+            StartupTime = DateTimeOffset.UtcNow;
+            BootTime = DateTimeOffset.UtcNow - TimeSpan.FromTicks(Stopwatch.GetTimestamp());
+        }
+
+        /// <summary>
+        /// By default, the StartupTime is set once SentryOptions is created.
+        /// That way is computationally cheap but not so precise.
+        /// You call this extension to give a better precision to the StartupTime at a cost of a small overhead to current
+        /// Thread.
+        /// </summary>
+        internal void SetupStartupTime()
```

Comment:
What about something like:
```suggestion
        /// </summary>
        internal void StartAccurateStartupTime()
```

Reply:
Indeed, StartupTime is already set, the original name doesn't make a lot of sense given the current setup.